src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java

```diff
@@ -712,6 +715,10 @@ else if (useFilteredViolations) {
             setFilteredViolation(inputConfigBuilder, lineNo + 1,
                     lines.get(lineNo), specifyViolationMessage);
         }
+        else if (violationsDefault.matches()) {
+            final int violationLineNum = lineNo + 1;
+            inputConfigBuilder.addViolation(violationLineNum, null);
```

Comment:
https://github.com/checkstyle/checkstyle/issues/15256#issuecomment-2233441811
> match all things violation, then figure out which it is, or throw an exception if it doesn’t match

This code makes it look like we capture this as a violation and continue on. I was expecting a blanket throw exception. Am I missing something? We prefer to see the violation message and not just the word violation by itself.

Reply:
We support violation by itself. There are a lot of Inputs with just `// violation `

Reply:
How did we support them when we just added this regexp? Show me we will get an exception with a violation comment that doesn't match our other regexps like issue talked about.

Reply:
we always supported it , and still allow if message is single in check - https://github.com/checkstyle/checkstyle/commit/a3c1783c90371f582d853d2e1d490f1ff847e30e#diff-7059762335dfd1e655cf558d06cfbe29db28808a2c154035b57c9d8dd2026740R10

but if people did typo , like extra space or some other, it did not match. What we do now is aggressive match of any "violation" word at any place to expected violation, so in this case we eliminating any possible typos or ...

Reply:
1) This doesn't change from the issue about `throw an exception if it doesn’t match`.
2) It doesn't explain how we handled `// violation` without the regular expression we just added now.
3) Why didn't this catch the issue's violation message.

Reply:
> This doesn't change from the issue about throw an exception if it doesn’t match

done.

Reply:
explanation is at https://github.com/checkstyle/checkstyle/pull/15021#discussion_r1657176223
I am posting here one more time.
![image](https://github.com/user-attachments/assets/46a460ec-d693-4aaa-a3ea-2b9d79a7eddc)
it is how regexp it written.

> If we supported bland violation matching, why didn't this catch the issue's violation message?

it was not bland, it was match to simple strict `// violation` or `// violation, 'message'` or  `// violation, "message"`